timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -148,13 +142,22 @@ def static_features(self, value: Optional[pd.DataFrame]):
         if not isinstance(self.index, pd.MultiIndex):
             return
 
-        if value is not None and not set(value.index).issuperset(set(self.item_ids)):
-            raise ValueError("Static features index should match item index")
-
-        # if static features being set are a strict superset of the item index, we take a
-        # subset to ensure consistency
-        if value is not None and len(set(value.index) - set(self.item_ids)) > 0:
-            value = value.loc[self.item_ids].copy()
+        if value is not None:
+            if isinstance(value, pd.Series):
+                value = value.to_frame()
+            if not isinstance(value, pd.DataFrame):
+                raise ValueError(f"static_features must be a pandas DataFrame (received object of type {type(value)})")
+            missing_item_ids = self.item_ids.difference(value.index)
+            if len(missing_item_ids) > 0:
+                raise ValueError(
+                    f"Following item_ids are missing from the index of static_features: {missing_item_ids.to_list()}"
+                )
+            # if provided static features are a strict superset of the item index, we take a subset to ensure consistency
+            if len(value.index.difference(self.item_ids)) > 0:
+                value = value.loc[self.item_ids].copy()
+            if value.index.name != ITEMID:
```

Comment:
why force the data to a convention we use instead of just replacing the splitter to take the first level of the index ?  especially because we seem to be taking yet another deep copy below just for this purpose.